src/legacy/components/analytics/hooks/useAnalyticsReporter.ts

```diff
@@ -90,6 +96,12 @@ export function useAnalyticsReporter() {
     }
   }, [account, walletName, prevAccount])
 
+  useEffect(() => {
+    // Custom dimension 5 - market
+    const market = getMarketDimension({ tradeTypeInfo, derivedTradeState })
+    googleAnalytics.setDimension(Dimensions.market, market)
+  }, [tradeTypeInfo, derivedTradeState])
```

Comment:
I think, we shouldn't call `setDimension` on every `derivedTradeState` change.
Since we need only tokens pair, we can di that:

```
const sellSymbol = derivedTradeState?.state?.inputCurrency?.symbol
const buySymbol = derivedTradeState?.state?.outputCurrency?.symbol
const tradePair = sellSymbol && buySymbol ? `${sellSymbol},${buySymbol}` : ''

useEffect(() => {
    // Custom dimension 5 - market
    const market = getMarketDimension({ tradeTypeInfo, tradePair })
    googleAnalytics.setDimension(Dimensions.market, market)
  }, [tradeTypeInfo, tradePair])
```

Reply:
Good suggestion